Microsoft.Toolkit.Uwp.SampleApp/Controls/SampleAppMarkdownRenderer.cs

```diff
@@ -73,53 +76,143 @@ protected override void RenderCode(CodeBlock element, IRenderContext context)
+
+                                // Remove old Viewer.
+                                var lastItem = prevPanel.Children.Count - 1;
+                                if (lastItem >= 0)
+                                {
+                                    prevPanel.Children.RemoveAt(lastItem);
+                                }
+
+                                prevPanel.Children.Add(newViewer);
+                            };
+
+                            LanguageRequested += (s, e) =>
+                            {
+                                if (s != combLangHead)
+                                {
+                                    if (combLangHead.Items.Contains(e))
+                                    {
+                                        combLangHead.SelectedItem = e;
+                                        block.CurrentLanguage = e;
+                                    }
+                                }
+                            };
+
+                            if (DesiredLang == language)
+                            {
+                                combLangHead.SelectedItem = language;
+                                block.CurrentLanguage = language;
+                            }
+                        }
+                        else if (langHead is ComboBox combLangHead)
+                        {
+                            // Add Lang to ComboBox
+                            block.Languages.Add(language, (viewer, element.Text));
+                            combLangHead.Items.Add(language);
 
-                copyButton.Click += (s, e) =>
+                            if (DesiredLang == language)
+                            {
+                                combLangHead.SelectedItem = language;
+                                block.CurrentLanguage = language;
+                            }
+                        }
+                    }
+                }
+                else
                 {
-                    var content = new DataPackage();
-                    content.SetText(element.Text);
-                    Clipboard.SetContent(content);
-                };
-
-                headerGrid.Children.Add(copyButton);
-                Grid.SetColumn(copyButton, 1);
-
-                // Collection the adornment and the standard UI, add them to a Stackpanel, and add it back to the collection.
-                var panel = new StackPanel();
-                panel.Children.Add(headerGrid);
-                panel.Children.Add(viewer);
-                panel.Background = viewer.Background;
-                panel.Margin = viewer.Margin;
-
-                collection.Add(panel);
+                    block = new CustCodeBlock();
+                    block.Languages.Add(language, (viewer, element.Text));
+                    block.CurrentLanguage = language;
+
+                    // Creates a Header to specify Language and provide a copy button.
```

Comment:
Feels like all these controls and things should be in a template resource which could be loaded/overridden.

Reply:
You mean just for the Sample App?

Having Templates for each Renderering type could be possible, it would just be a lot to implement.

Reply:
I mean whoever uses the markdown control may want to change the styling of the header of the code sample to fit their app's theming or style, etc...  With all this UI stuff being hardcoded they can't do that.  It doesn't have to be a complex template but something like:

```
<Grid>
    <Grid.RowDefinitions>
        <RowDefinition Height="Auto"/>
        <RowDefinition/>
    </Grid.RowDefinitions>
    <Grid.ColumnDefinitions>
        <ColumnDefinition/>
        <ColumnDefinition Width="Auto"/>
    </Grid.ColumnDefinitions>
    <ComboBox x:Name="LanguageDropdown"/>
    <Button x:Name="CopyButton" Grid.Column="1" Content="Copy"/>
    <ContentPresenter Grid.Row="1" Grid.ColumnSpan="2"/>
</Grid>
```

Reply:
This is entirely in the `SampleAppMarkdownRenderer`, so it only affects the Sample App.

Reply:
Ah, missed that, no worried then.